src/story_protocol_python_sdk/resources/IPAsset.py

```diff
@@ -174,3 +178,114 @@ def registerDerivativeWithLicenseTokens(self, child_ip_id: str, license_token_id
 
         except Exception as e:
             raise e
+
+    def mintAndRegisterIpAssetWithPilTerms(self, nft_contract: str, pil_type: str, metadata: dict = None, recipient: str = None, minting_fee: int = None, commercial_rev_share: int = None, currency: str = None, tx_options: dict = None) -> dict:
+        """
+        Mint an NFT from a collection and register it as an IP.
+
+        :param nft_contract str: The address of the NFT collection.
+        :param pil_type str: The type of the PIL.
+        :param metadata dict: [Optional] The metadata for the IP.
+            :param metadataURI str: [Optional] The URI of the metadata for the IP.
+            :param metadataHash str: [Optional] The metadata hash for the IP.
+            :param nftMetadataHash str: [Optional] The metadata hash for the IP NFT.
+        :param recipient str: [Optional] The address of the recipient of the minted NFT.
+        :param minting_fee int: [Optional] The fee to be paid when minting a license.
+        :param commercial_rev_share int: [Optional] Percentage of revenue that must be shared with the licensor.
+        :param currency str: [Optional] The ERC20 token to be used to pay the minting fee. The token must be registered in Story Protocol.
+        :param tx_options dict: [Optional] The transaction options.
+        :return dict: A dictionary with the transaction hash.
+        """
+        try:
+            if pil_type is None or pil_type not in PIL_TYPE.values():
+                raise ValueError("PIL type is required and must be one of the predefined PIL types.")
+
+            if not self.web3.is_address(nft_contract):
+                raise ValueError(f"The NFT contract address {nft_contract} is not valid.")
+
+            license_term = get_license_term_by_type(pil_type, {
+                'mintingFee': minting_fee,
+                'currency': currency,
+                'royaltyPolicy': "0xAAbaf349C7a2A84564F9CC4Ac130B3f19A718E86", 
```

Comment:
where does this address come from? best not to hard code things -- should it be the default option?